src/orange/OrangeData.hh

```diff
@@ -26,7 +26,7 @@ namespace celeritas
  */
 struct OrangeParamsScalars
 {
-    static constexpr size_type max_level{1};
+    static constexpr size_type max_level{20};
```

Comment:
This should be a problem-dependent property since we can allocate all the states at once:
```suggestion
    size_type max_level{};
```

Reply:
I did it this way because it's a static constexpr within Shift ORANGE. I agree this seems like it wastes a lot of memory though. If we make it problem-dependent, the LevelStateAccessor is going to need to take `max_level` as an argument every time it's constructed though, right?

Reply:
I thought we discussed this and how it's different because of the memory models between shift and orange (one allocation per particle vs all particles). Good point about the LSA but I think it's worth it. Maybe make LSA take `const OrangeParams&` . And a helper function in the TrackView to emit an LSA for a given level would help out greatly.

Reply:
What if we pass max_level to LSA an then add helper functions to OrangeTrackView to do so? I am concerned about the amount of indirection that is going on w.r.t. the LSA, since really all it doing is the very simple operation `index = max_level*thread + level`.

Reply:
LSA could still store `max_level` locally, I just meant that using `OrangeParamsData` as the constructor argument will make it easer to refactor if needed. I think passing max_level through the constructor is also fine as long as we have the helper function.

Reply:
Now that I am working through this, I actually think `max_level` should be additionally stored in `OrangeStateData` as a `stride` or `dims` member variable. Otherwise there is no way of accessing the the flattened 2D data (`pos`, `dir`, etc) without an `OrangeParamsData` object. This isn't so bad within `OrangeTrackView` because there is an `OrangeParamData` object handy.  However, `EnergyDiagnosticLauncher` does not have access to an `OrangeParamsData` object and there probably isn't a good way of getting it the `max_level` information any other way than via `OrangeStateData` without API changes.

It would be nice to at some point have a `HyperslabCollection` class in which case the stride would be stored under the hood, but until then I think storing the stride next to the data itself is the simplest path forward. This simplifies the LSA as well.

Reply:
The energy diagnostic launcher is a total hack and shouldn't be factored into any of our long-term design plans. Overall the intent of the Params/State is to completely separate problem-dependent parameters from particle-dependent state, so it's intentional that we always need to use a composition of data from both the parameters and the state.

That being said, if putting a copy of `max_level` into the state is going to save us a lot of tedium or refactoring in the short term, let's just do that (it's not like the single integer is going to cause us memory problems) and annotate it with
```c++
// TODO: this is problem-dependent data and should eventually be removed
```
or something.